adal/src/main/java/com/microsoft/aad/adal/BrokerAccountServiceHandler.java

```diff
@@ -151,7 +151,16 @@ public void onError(Throwable throwable) {
 
         final Throwable throwable = exception.getAndSet(null);
         if (throwable != null) {
-            throw new AuthenticationException(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, throwable.getMessage(), throwable);
+            if (throwable instanceof RemoteException) {
+                Logger.e(TAG, "Get error when trying to get token from broker: " + throwable.getMessage(), "", ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, throwable);
+                throw new AuthenticationException(ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, throwable.getMessage(), throwable);
+            } else if (throwable instanceof InterruptedException) {
+                Logger.v(TAG, "The thread is interrupted while it is sleeping. "  + throwable.getMessage());
+                //Should it throw the InterruptedException?
```

Comment:
It's not clear to me when this Exception will be thrown (`InterruptedException`) or indeed if it will ever be thrown at all -- my gut tells me we should probably throw this Exception 'up' rather than allow execution to proceed with a `null` result.

Reply:
And I am not quite sure what should we do with the InterruptedException. I checked across our repos, and we just log the InterruptedException in verbose level.